File: Medical-Insurance/Com.Application.Domain.Entities/Familymember.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Com.Application.Domain.Entities;

public partial class Familymember : BaseEntity
{
    public int MemberId { get; set; }

    public int PolicyId { get; set; }

    public string MemberName { get; set; } = null!;

    public string Relation { get; set; } = null!;

    public int? Age { get; set; }

    public string? Gender { get; set; }

    //public virtual Policy Policy { get; set; } = null!;
    [JsonIgnore]
    public virtual Policy? Policy { get; set; } 

}

File: clearclaim-frontend/src/models/Familymember.ts
// Familymember.ts
// Represents the FamilyMember entity from the backend
// Maps to the familymember table in PostgreSQL

export interface Familymember {
  memberId: number;
  policyId: number;
  memberName: string;
  relation: string;
  age?: number;
  gender?: string;
}
